### PR TITLE
Added an east north up frame object

### PR DESCRIPTION
**Pull request by Joshuaalbert**
This contains a new representation (ENU) for the East-North-Up (right handed) system at a given observer location and obstime.
It gives the transformations between ENU and the ITRS frames for the graph transformation.
The use case I originally wrote this for was for ray tracing.

### CODE REVIEW

File: astropy/coordinates/builtin_frames/enu.py [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+                                     y = penu[1],#u.Quantity(penu[1],u.m,copy=False),
+                                     z = penu[2],#u.Quantity(penu[2],u.m,copy=False),
+                                     copy=False)
+
+    return enu_frame.realize_frame(rep)
+
+
+@frame_transform_graph.transform(FunctionTransform, ENU, ITRS)
+def enu_to_itrs(enu_coo, itrs_frame):
+    #p = itrs_frame.cartesian.xyz.to(u.m).value
```
Comment (pllim) [inline astropy/coordinates/builtin_frames/enu.py]: Again, delete unused variables? Also, no docstring.

File: astropy/coordinates/builtin_frames/enu.py [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+    
+    #return ITRS(*p*u.m,obstime=enu_coo.obstime).transform_to(itrs_frame)
+    
+@frame_transform_graph.transform(FunctionTransform, ENU, ENU)
+def enu_to_enu(from_coo, to_frame):
+    # for now we just implement this through ITRS to make sure we get everything
+    # covered
+    return from_coo.transform_to(ITRS(obstime=from_coo.obstime,location=from_coo.location)).transform_to(to_frame)
+
+if __name__ == '__main__':
```
Comment (pllim) [inline astropy/coordinates/builtin_frames/enu.py]: Move test to a proper test module that can be discovered and run by pytest.

**Comment by Joshuaalbert** [comment]
Addressed the comments.

**Comment by Joshuaalbert** [comment]
Okay all checks passed now and your comments implemented.

**Comment by bsipocz** [comment]
> @pllim Okay all checks passed now and your comments implemented.

@Joshuaalbert - This will definitely need unit tests and documentation, but probably it's OK to wait for a review from the module maintainers first.

**Comment by eteq** [comment]
Note that this closes #5300

**Comment by astrofrog** [comment]
@eteq - if you are able to review this, do you want to re-open?

**Comment by astrofrog** [comment]
This was re-opened as https://github.com/astropy/astropy/pull/6882

**Pull request closed without merging**